libs/x-content/src/main/java/org/elasticsearch/common/xcontent/MediaTypeParser.java

```diff
@@ -96,6 +91,17 @@ public ParsedMediaType parseMediaType(String headerValue) {
         return null;
     }
 
+    private boolean isValidParameter(String typeWithSubtype, String parameterName, String parameterValue) {
+        if(parametersMap.containsKey(typeWithSubtype)){
+            Map<String, String> parameters = parametersMap.get(typeWithSubtype);
+            if(parameters.containsKey(parameterName)){
+                String regex = parameters.get(parameterName);
+                return parameterValue.matches(regex);//todo pg should we precompile regex?
```

Comment:
We should compile the regex into a pattern to avoid doing it everytime

Reply:
will do